### DISCUSSION TITLE
Inject linkerd in namespaces

### DISCUSSION THREAD

**Discussion by shahriak** [linkerd/linkerd2#6100] [Q&A] [answered]
Hello everyone,

Is there a way to inject linkerd on a namespace so that all resources deployed in that namespace are automatically injected with linkerd proxy?
I was not able find this in the docs: https://linkerd.io/2.10/reference/cli/inject/

**Comment by mateiidavid** [comment, marked as answer]
Yup! If you add the inject annotation (`linkerd.io/inject: enabled`) on a namespace then all workloads in that namespace will be injected.

**Reply by olix0r** [reply]
There's some more information about this here https://linkerd.io/2.10/features/proxy-injection/

**Reply by shahriak** [reply]
> Yup! If you add the inject annotation (`linkerd.io/inject: enabled`) on a namespace then all workloads in that namespace will be injected.

@mateiidavid  @olix0r I have actually tested this out. Injecting the namespace with Linkerd (kubectl get ns -o yaml | linkerd inject - | kubectl apply -f -)  and then deploying the app in the namespace does not automatically injects a sidecar in the pods. I had to inject linkerd in the deployment yaml explicitly in order for that to happen. Is there a plan to add namespace injection sometime in the future?

**Reply by mateiidavid** [reply]
Hey @shahriak, perhaps something is misconfigured?

I managed to get this working with our `emojivoto` example app, I'm running the latest edge (`edge-21.4.5`).
```
apiVersion: v1
kind: Namespace
metadata:
  name: emojivoto
  annotations:
    linkerd.io/inject: enabled
```

If I create the namespace above and then create deployments inside the namespace (without any additional inject annotations), the sidecar proxies get injected as expected.

Are you seeing any errors or anything similar? Namespace injection should work with no configuration on your part.

**Reply by shahriak** [reply]
> Hey @shahriak, perhaps something is misconfigured?
> 
> I managed to get this working with our `emojivoto` example app, I'm running the latest edge (`edge-21.4.5`).
> 
> ```
> apiVersion: v1
> kind: Namespace
> metadata:
>   name: emojivoto
>   annotations:
>     linkerd.io/inject: enabled
> ```
> 
> If I create the namespace above and then create deployments inside the namespace (without any additional inject annotations), the sidecar proxies get injected as expected.
> 
> Are you seeing any errors or anything similar? Namespace injection should work with no configuration on your part.

I am actually using the stable 2.10.0 version. I have checked that the annotation exists in the namespace as well..

**Reply by mateiidavid** [reply]
Have you rolled your deployments (or applied them) after the namespace was annotated? Namespace annotation needs to be in place before the pods are created for the [`proxy-injector`](https://linkerd.io/2.10/features/proxy-injection/) to inject the sidecar proxies.

Another issue could be in your deployments if they contain `linkerd.io/inject: disabled` as an annotation in the pod spec.

**Reply by shahriak** [reply]
Yup. That is the issue, the namespace is being reconfigured after I execute this command: curl -sL https://run.linkerd.io/emojivoto.yml | kubectl -n emojivoto apply -f - to deploy emojivoto app in the namespace
